### PR TITLE
Fix notice styles

### LINKED ISSUE #42690: Notices appear in old styles after blocks monorepo migration
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In p1702294257650489/1702031295.900509-slack-C02UBB1EPEF, @jimjasson reported that the notices are showing the old styles when adding a product to the cart. This was detected due to a failing e2e test that ran against the nightly build of Woo core.

### Expected behavior

When adding a product to the cart, the new notice style should be visible.

### Actual behavior

When adding a product to the cart, the old notice style should be visible.

### Steps to reproduce

1. Go to a product page.
2. Add the product to the cart.
3. See that the old notice style is visible instead of the new one.

### WordPress Environment

<details>
<summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): https://store.test
Site address (URL): https://store.test
WC Version: 8.3.0
REST API Version: ✔ 8.3.0
WC Blocks Version: ✔ 11.8.0-dev
Action Scheduler Version: ✔ 3.7.0
Log Directory Writable: ✔
WP Version: 6.4.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.25.3
PHP Version: 8.1.26
PHP Post Max Size: 512 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 8.4.0
(SecureTransport) OpenSSL/3.2.0

SUHOSIN Installed: –
MySQL Version: 8.2.0
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.3.0
WC Database Prefix: wp_
Total Database Size: 32.12MB
Database Data Size: 29.18MB
Database Index Size: 2.94MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.14MB + Index: 0.14MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.06MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 27.05MB + Index: 0.08MB + Engine InnoDB
wp_postmeta: Data: 0.13MB + Index: 0.09MB + Engine InnoDB
wp_posts: Data: 0.16MB + Index: 0.06MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.05MB + Index: 0.13MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.08MB + Index: 0.14MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wps_hit: Data: 0.02MB + Index: 0.14MB + Engine InnoDB
wp_wps_index: Data: 0.17MB + Index: 0.20MB + Engine InnoDB
wp_wps_key: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wps_object_term: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wps_object_type: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_wps_query: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wps_uri: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wps_user_agent: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_indexable: Data: 0.05MB + Index: 0.09MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.03MB + Engine InnoDB

### Post Type Counts ###

attachment: 23
page: 12
post: 16
product: 18
product_variation: 7
revision: 14
shop_coupon: 1
shop_order: 5
wp_global_styles: 3
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (5) ###

Another Show Hooks: by Exlac – 1.0.2
Code Snippets: by Code Snippets Pro – 3.6.2
SMNTCS Show Symlinked Plugins: by Niels Lange – 1.3
SMNTCS Theme List View: by Niels Lange – 1.3
WooCommerce: by Automattic – 8.5.0-dev

### Inactive Plugins (25) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3
Checkout Filters for WooCommerce Blocks: by Niels Lange – 1.0.0
Coinbase Commerce: by Coinbase Commerce – 1.3
Gutenberg: by Gutenberg Team – 17.2.0
Mercado Pago payments for WooCommerce: by Mercado Pago – 7.0.6
Paystack WooCommerce Payment Gateway: by Tunbosun Ayinla – 5.8.1
PayU India: by PayU – 3.8.2
Query Monitor: by John Blackbourn – 3.15.0
Razorpay for WooCommerce: by Team Razorpay – 4.5.7
Simple Custom CSS and JS: by SilkyPress.com – 3.45
VAT field for WooCommerce Blocks: by Niels Lange – 1.0.0
WooCommerce Authorize.Net Gateway: by SkyVerge – 3.9.3
WooCommerce Bookings: by WooCommerce – 2.0.3
WooCommerce Dummy Payments Gateway: by SomewhereWarm – 1.0.5
WooCommerce EU VAT Number: by WooCommerce – 2.8.8
Woocommerce Free Shipping Progress Bar Block: by Automattic – 0.1.0
WooCommerce Payfast Gateway: by WooCommerce – 1.6.0
WooCommerce PayPal Payments: by WooCommerce – 2.4.2
WooCommerce Product Search: by itthinx – 4.16.0
WooCommerce Shipment Tracking: by WooCommerce – 2.3.1
WooCommerce Stripe Gateway: by WooCommerce – 7.7.0
WooCommerce Subscriptions: by WooCommerce – 5.1.2
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.5.5
WordPress Importer: by wordpressdotorg – 0.8.1
WP User Role Switcher: by Dhanendran Rajagopal – 0.2.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: ✔

### WC Pages ###

Shop base: #2 - /products-beta/classic-shop/
Cart: #6 - /cart/classic-cart-shortcode/
Checkout: #8 - /checkout/classic-checkout-shortcode/
My account: #10 - /my-account/
Terms and conditions: #11 - /terms/

### Theme ###

Name: Storefront
Version: 4.5.3
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: /Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/notices/error.php
/Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/notices/notice.php
/Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-external-affiliate
product-grouped
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2023-12-12 13:49:42 +00:00
Options: ✔
Notes: 90
Onboarding: skipped

### Action Scheduler ###

Canceled: 1
Oldest: 2023-12-11 11:56:59 +0000
Newest: 2023-12-11 11:56:59 +0000

Complete: 74
Oldest: 2023-11-13 04:59:42 +0000
Newest: 2023-12-12 08:22:23 +0000

Failed: 17
Oldest: 2023-05-15 07:30:15 +0000
Newest: 2023-11-29 14:59:20 +0000

Pending: 2
Oldest: 2023-12-12 11:54:36 +0000
Newest: 2023-12-24 11:10:56 +0000


### Status report information ###

Generated at: 2023-12-12 08:41:47 +00:00
```

</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by nielslange**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #42690.

In p1702294257650489/1702031295.900509-slack-C02UBB1EPEF, @jimjasson reported that the notices are showing the old styles when adding a product to the cart. This was detected due to a failing e2e test that ran against the nightly build of Woo core.

Initially, we thought that this issue was caused by https://github.com/woocommerce/woocommerce-blocks/pull/12043. While investigating this issue, I noticed that while migrating WooCommerce Blocks into core, the folder with the notice templates had been renames from `notices` to `block-notices`. This change took place in [https://github.com/woocommerce/woocommerce/commit/6950115000d562f53271f719a7c2070ffce4c4f9#diff-04997489ff1d17d1c6f2a1[…]f40a1ee468ba2c6a58be46dab10](https://github.com/woocommerce/woocommerce/commit/6950115000d562f53271f719a7c2070ffce4c4f9#diff-04997489ff1d17d1c6f2a147418afe91041c0f40a1ee468ba2c6a58be46dab10). 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Testing fixed notices

1. Install and activate the [Twenty Twenty-Four theme](https://wordpress.org/themes/twentytwentyfour/).
2. Create a test page and add the cart shortcode to it.
3. Go to the frontend and visit a product page.
4. Add the product to the cart.
5. Verify that the new notice style is visible.

<table>
<tr>
<td valign="top">Old notice style:
<br><br>
Screenshot 2023-12-12 at 16 33 09
</td>
<td valign="top">New notice style:
<br><br>
Screenshot 2023-12-12 at 16 30 36
</td>
</tr>
</table>

6. Go to the test page with the cart shortcode.
7. Empty all form fields and submit the checkout form.
8. Verify that the new notice style is visible.

<table>
<tr>
<td valign="top">Old notice style:
<br><br>
Screenshot 2023-12-15 at 13 38 03
</td>
<td valign="top">New notice style:
<br><br>
Screenshot 2023-12-15 at 13 35 47
</td>
</tr>
</table>

> [!IMPORTANT]
> Please note that when using the Storefront theme, the SVG icon before the notice text is not visible. This issue should be addressed in a separate PR as it only affects the Storefront theme.

#### Testing existing notices

1. Go to `/my-account/`.
2. Log out, if you're currently logged in.
3. Simulate a login, but provide an incorrect username or password.
4. Verify that the error message is still working, and using the new style.
6. Log in and go to `/my-account/edit-address/`.
10. Edit the billing or shipping address.
11. Verify that the success message is still working, and using the new style.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [x] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Fix: Ensure that new notice styles are visible.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/templates/notices/success.php [hunk 1]
```diff
@@ -26,7 +26,12 @@
...
 <?php foreach ( $notices as $notice ) : ?>
-	<div class="woocommerce-message"<?php echo wc_get_notice_data_attr( $notice ); ?> role="alert">
-		<?php echo wc_kses_notice( $notice['notice'] ); ?>
+	<div class="wc-block-components-notice-banner is-success"<?php echo wc_get_notice_data_attr( $notice ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?> role="alert">
+		<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" aria-hidden="true" focusable="false">
+			<path d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"></path>
+		</svg>
+		<div class="wc-block-components-notice-banner__content">
+			<?php echo wc_kses_notice( $notice['notice'] ); ?>
+		</div>
```
Comment (senadir) [inline plugins/woocommerce/templates/notices/success.php:35]: Can you help walk me thought those changes Niels, we merged existing notices templates? would this apply to all themes or just block themes?

**Review by senadir — APPROVED** [review]
This is testing fine to me. left a question to understand more but it's not a blocker. I also rebased to trunk.

File: plugins/woocommerce/templates/notices/success.php (hunk 1 above)
Comment (nielslange) [reply plugins/woocommerce/templates/notices/success.php:35]: Thanks for asking, @senadir. This change affects all themes, both classic and block themes. This change had been requested in https://github.com/woocommerce/woocommerce-blocks/issues/12042 and implemented in https://github.com/woocommerce/woocommerce-blocks/pull/12043. Due to the merge, we've accidentally switched back to the old notices. This PR can be considered an improved version of https://github.com/woocommerce/woocommerce-blocks/pull/12043.

**Comment by JimmyAppelt** [comment]
@jonathansadowski @nielslange hi there 🙂 

- There is something wrong with the notice template overriding now, likely due to the implementation and how templates are called in the `Automattic\WooCommerce\Blocks\Domain\Services\Notices` class.
- Is the `woo-gutenberg-products-block domain correct here?`
- The template versions are still on `-dev`

https://github.com/woocommerce/woocommerce/issues/43342

**Comment by nielslange** [comment]
Thanks for reaching out and reporting these issues in #43342, @JimmyAppelt.

**Comment by DarkChris86** [comment]
Dear @nielslange, @jonathansadowski, @JimmyAppelt,

This problem is persists in the latest 'stable' 8.5.0 version also.
I started a support ticket [here](https://wordpress.org/support/topic/wordpress-8-5-issue-wocommerce-notices-template-files-has-no-effect/#post-17335153) about it.

**Comment by nielslange** [comment]
Thanks for reaching out, @DarkChris86. I was able to reproduce this issue and addressed it in #43506. The fix should be shipped with the next patch release.

**Comment by DarkChris86** [comment]
Great news @nielslange!
Thanks for your contribution.

**Comment by nielslange** [comment]
Thanks for reaching out to report this, @DarkChris86. 🙌

**Pull request merged**